src/svelte5.svelte.js

```diff
@@ -0,0 +1,49 @@
+/** eslint-global: $state */
+import { mount, unmount } from 'svelte'
+
+import { SvelteTestingLibrary } from './pure.js'
+
+class Svelte5TestingLibrary extends SvelteTestingLibrary {
+  svelteComponentOptions = [
+    'target',
+    'props',
+    'events',
+    'context',
+    'intro',
+    'recover',
+  ]
+
+  propsByComponent = new Map()
+
+  renderComponent(ComponentConstructor, componentOptions) {
+    const props = $state(componentOptions.props ?? {})
+    const component = mount(ComponentConstructor, {
+      ...componentOptions,
+      props,
+    })
+
+    this.componentCache.add(component)
+    this.propsByComponent.set(component, props)
+
+    return component
+  }
+
+  rerenderComponent(component, nextProps) {
+    const prevProps = this.propsByComponent.get(component)
+    Object.assign(prevProps, nextProps)
+  }
+
+  cleanupComponent(component) {
+    const inCache = this.componentCache.delete(component)
+    this.propsByComponent.delete(component)
+
+    if (inCache) {
+      unmount(component)
+    }
+  }
+}
+
+const instance = new Svelte5TestingLibrary()
+
+export const render = instance.render.bind(instance)
+export const cleanup = instance.cleanup.bind(instance)
```

Comment:
@yanick I find this singleton-class-instance-for-deduping-via-inheritance a bit confusing. I think this could be implemented more cleanly and directly via more procedural function composition. Would you be open to such a change, especially to merge `svelte5` back into `pure` for the Svelte 5 production release?

Reply:
It depends a lot what that "more cleanly and directly" looks like. For what it's worth, when svelte5 will become the default, I was not going to merge it back into `pure` but rather have the default point to it, so that

`import {} from 'svelte-testing-library' ;` gives you the official live svelte version of the day.
`import {} from 'svelte-testing-library/svelte4' ;` gives you the svelte 4 version
`import {} from 'svelte-testing-library/svelte5' ;` gives you the svelte 5 version

Reply:
> It depends a lot what that "more cleanly and directly" looks like

Given the small scope of this library, I think something that more obviously (and, to be honest, crudely) maps out the branches as you read would be easier to follow as a reader:

```js
// pure.js

export const render = (...) => {
  // ...
  const component = IS_SVELTE_5
    ? renderSvelte5Component(...)
    : renderLegacyComponent(...)
  // ...
}

export const cleanup = () => {
  componentCache.forEach(cleanupComponent)
  // ...
}

const cleanupComponent = (component) => {
  if (IS_SVELTE_5) {
    cleanupSvelte5Component(component)
  } else {
    cleanupLegacyComponent(component)
  }
}
```

> I was not going to merge it back into pure but rather have the default point to it, so that
> 
> import {} from 'svelte-testing-library' ; gives you the official live svelte version of the day.
> import {} from 'svelte-testing-library/svelte4' ; gives you the svelte 4 version
> import {} from 'svelte-testing-library/svelte5' ; gives you the svelte 5 version

Have you read through the inbound issues on #284? The separate entry points have been leading to relatively frequent cleanup issues and general confusion. Splitting the library into separate singleton instances with independent state has been a little rough. Plus it's been hard to get the APIs correct on our end as authors: see #346, #339. 

I don't think it's a pattern we should keep going with. I don't see any technical reason to continue with separate entry points given that the `svelte/legacy` import is no longer needed

Reply:
> export const render = (...) => {
>  // ...
>  const component = IS_SVELTE_5
>    ? renderSvelte5Component(...)
>   : renderLegacyComponent(...)
> // ...
> }

My original reaction is 'oh sweet baby jesus no'. Readability is often in the eye in the beholder, and mushing things together like that is pretty much my definition of how not to do it. That doesn't get my vote, sorry.

> . I don't see any technical reason to continue with separate entry points given that the svelte/legacy import is no longer needed

We might have different expectations there. Even when svelte5 is release, I expect the testing library to still support svelte 4 as long as it doesn't get too onerous to do it.

Reply:
> My original reaction is 'oh sweet baby jesus no'. Readability is often in the eye in the beholder, and mushing things together like that is pretty much my definition of how not to do it. That doesn't get my vote, sorry.

My example was intentionally crude for the purposes of a GitHub comment. What I'm proposing is continuing to use the breakdown of tasks that the classes use, but simply export functions instead of using singletons and `bind`s everywhere. There are real technical downsides and footguns to the class-based approach that we've been dealing with.

Regardless, I'd like us to come up with a solution that doesn't split the cleanup cache up, which seems to be the main driver of issues that users have been encountering. 

> > I don't see any technical reason to continue with separate entry points given that the svelte/legacy import is no longer needed
>
> We might have different expectations there. Even when svelte5 is release, I expect the testing library to still support svelte 4 as long as it doesn't get too onerous to do it.

We have the same expectations, this library should continue to support the versions of Svelte it works with today. What I'm saying is that this library doesn't need separate entry-points to do so. A single entry-point can easily support Svelte 3, 4 and 5

As a user of this library, I find the separate entry-points to be a weird bit of busy work to force on me as I upgrade. I think as library authors we should strive to force as little upgrade work on our users as possible

Reply:
Threw up a little WIP for what's on my mind via no singletons + single entrypoint:

https://github.com/testing-library/svelte-testing-library/compare/svelte5-no-legacy...testing-core-mod

Reply:
> My example was intentionally crude for the purposes of a GitHub comment.

Then you can understand how it doesn't do a super job of selling your point.

Now, I took maintainership of this project because it was needed and, well, because it was fun. But there are a lot of other balls I have to juggle outside of this project, and I'm afraid you're pulling too fast and too much in a different direction for this old man to keep up. So it makes sense for me to take a step back and a little sabbatical. You already have the keys to the car. So go ahead and do what thou wilt. And best of luck. :-)

Reply:
Gotcha! I appreciate you keeping the project going. I'll keep an eye out for any feedback whenever you feel like dropping by